### ISSUE TITLE
Training epoch loss and validation epoch loss are calculated wrong in Trainer

### ISSUE THREAD

**Issue by longdht** [awslabs/gluon-ts#1684]
## Description
Since Both loop() for training epoch and validation epoch are assigned as epoch_loss, when training with both training data and validation data, the epoch loss used in callbacks have no meaning.
It should be assigned with training_epoch_loss and validation_epoch_loss separately.

## To Reproduce
In gluon-ts/src/gluonts/mx/trainer/_base.py:
In  def loop:

```python
                epoch_loss = loop(
                    epoch_no,
                    train_iter,
                    num_batches_to_use=self.num_batches_per_epoch,
                )

                should_continue = self.callbacks.on_train_epoch_end(
                    epoch_no=epoch_no,
                    epoch_loss=loss_value(epoch_loss),
                    training_network=net,
                    trainer=trainer,
                )

                if is_validation_available:
                    epoch_loss = loop(
                        epoch_no, validation_iter, is_training=False
                    )

                    should_continue = (
                        should_continue
                        and self.callbacks.on_validation_epoch_end(
                            epoch_no=epoch_no,
                            epoch_loss=loss_value(epoch_loss),
                            training_network=net,
                            trainer=trainer,
                        )
                    )

## Error message or code output
no error message, only when using custom callbacks I had some issues due to wrong calculation

```
put error or undesired output here
```

## Environment
- Operating system: tried on mac m1 and linux
- Python version: tried with 3.8 and
- GluonTS version: 0.8.1
- MXNet version: 1.6.0

**Comment by lostella** [comment]
@longdht note that the snippet you posted is not from within the definition of `loop`.

I do think the code would be less confusing if `loop` would be defined in at the module’s top level, but I don’t see the bug here: `epoch_loss` takes a certain value first, which is given to the `on_train_epoch_end` callback, and then another value, which is passed to the `on_validation_epoch_end`. That seems ok to me, but maybe I misunderstood your point.

**Comment by longdht** [comment]
@lostella thank you for your comment,
ah yes its not in the loop, that was my mistake

Sorry this is my first time reporting issues on github so I think I didnt explain well
I was saying that it is better to call it by different name: training_epoch_loss and validation_epoch_loss, otherwise it will confuse people, and also on many callbacks we use epoch_loss, so we should know exactly which epoch_loss we are talking about.

Here is the problem
on the callbacks.on_epoch_end():
when we train without validation data, loss_value(epoch_loss) here is training_epoch_loss, while if we train with validation data, loss_value(epoch_loss) becomes validation_epoch_loss. It should not be confusing like that in my opinion

```
                should_continue = (
                    should_continue
                    and self.callbacks.on_epoch_end(
                        epoch_no=epoch_no,
                        epoch_loss=loss_value(epoch_loss),
                        training_network=net,
                        trainer=trainer,
                        best_epoch_info=best_epoch_info,
                        ctx=self.ctx,
                    )
                )
```

**Comment by longdht** [comment]
I just tried to explain the issue in more detailed, hope I can get some comment from everyone
